src/atomate2/common/schemas/pheasy.py

```diff
@@ -0,0 +1,812 @@
+"""Schemas for phonon documents."""
+
+import copy
+import logging
+from pathlib import Path
+from typing import Optional, Union
+
+import numpy as np
+from emmet.core.math import Matrix3D
+from emmet.core.structure import StructureMetadata
+from monty.json import MSONable
+from phonopy import Phonopy
+from phonopy.phonon.band_structure import get_band_qpoints_and_path_connections
+from phonopy.structure.symmetry import symmetrize_borns_and_epsilon
+from phonopy.units import VaspToTHz
+from pydantic import BaseModel, Field
+from pymatgen.core import Structure
+from pymatgen.io.phonopy import (
+    get_ph_bs_symm_line,
+    get_ph_dos,
+    get_phonopy_structure,
+    get_pmg_structure,
+)
+from pymatgen.io.vasp import Kpoints
+from pymatgen.phonon.bandstructure import PhononBandStructureSymmLine
+from pymatgen.phonon.dos import PhononDos
+from pymatgen.phonon.plotter import PhononBSPlotter, PhononDosPlotter
+from pymatgen.symmetry.bandstructure import HighSymmKpath
+from pymatgen.symmetry.kpath import KPathSeek
+from typing_extensions import Self
+
+# import lib by jiongzhi zheng 
+from ase.io import read
+from hiphive import ClusterSpace, ForceConstantPotential, ForceConstants, enforce_rotational_sum_rules
+from hiphive.cutoffs import estimate_maximum_cutoff
+from hiphive.utilities import extract_parameters
+import subprocess
+from phonopy.file_IO import write_FORCE_CONSTANTS, parse_FORCE_CONSTANTS
+from phonopy.interface.vasp import write_vasp
+from phonopy.interface.vasp import read_vasp
+
+
+from atomate2.aims.utils.units import omegaToTHz
+
+logger = logging.getLogger(__name__)
+
+
+def get_factor(code: str) -> float:
+    """
+    Get the frequency conversion factor to THz for each code.
+
+    Parameters
+    ----------
+    code: str
+        The code to get the conversion factor for
+
+    Returns
+    -------
+    float
+        The correct conversion factor
+
+    Raises
+    ------
+    ValueError
+        If code is not defined
+    """
+    if code in ["forcefields", "vasp"]:
+        return VaspToTHz
+    if code == "aims":
+        return omegaToTHz  # Based on CODATA 2002
+    raise ValueError(f"Frequency conversion factor for code ({code}) not defined.")
+
+
+class PhononComputationalSettings(BaseModel):
+    """Collection to store computational settings for the phonon computation."""
+
+    # could be optional and implemented at a later stage?
+    npoints_band: int = Field("number of points for band structure computation")
+    kpath_scheme: str = Field("indicates the kpath scheme")
+    kpoint_density_dos: int = Field(
+        "number of points for computation of free energies and densities of states",
+    )
+
+
```

Comment:
I think you can simply reuse from phonons.py in many cases. Please instead simply import.

Reply:
I import most of the class methods directly from schemas/phonons.py.